src/System.Text.Json/src/System/Text/Json/JsonUtf8Reader.cs

```diff
@@ -0,0 +1,1274 @@
+            if (i < data.Length)
+            {
+                nextByte = data[i];
+                if (JsonConstants.Delimiters.IndexOf(nextByte) >= 0)
+                    return ConsumeNumberResult.Success;
+            }
+            else
+            {
+                if (IsLastSpan)
+                    return ConsumeNumberResult.Success;
+                else
+                    return ConsumeNumberResult.NeedMoreData;
+            }
+            nextByte = data[i];
+            if (nextByte != '.' && nextByte != 'E' && nextByte != 'e')
+            {
+                _lineBytePosition += i;
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedNextDigitComponentNotFound, nextByte);
+            }
+
+            return ConsumeNumberResult.OperationIncomplete;
+        }
+
+        private ConsumeNumberResult ConsumeIntegerDigits(ref ReadOnlySpan<byte> data, ref int i)
+        {
+            byte nextByte = default;
+            for (; i < data.Length; i++)
+            {
+                nextByte = data[i];
+                if ((uint)(nextByte - '0') > '9' - '0')
+                    break;
+            }
+            if (i >= data.Length)
+            {
+                if (IsLastSpan)
+                    return ConsumeNumberResult.Success;
+                else
+                    return ConsumeNumberResult.NeedMoreData;
+            }
+            if (JsonConstants.Delimiters.IndexOf(nextByte) >= 0)
+                return ConsumeNumberResult.Success;
+
+            return ConsumeNumberResult.OperationIncomplete;
+        }
+
+        private ConsumeNumberResult ConsumeDecimalDigits(ref ReadOnlySpan<byte> data, ref int i, byte nextByte)
+        {
+            if (i >= data.Length)
+            {
+                if (IsLastSpan)
+                {
+                    _lineBytePosition += i;
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedDigitNotFoundEndOfData, nextByte);
+                }
+                else
+                    return ConsumeNumberResult.NeedMoreData;
+            }
+            nextByte = data[i];
+            if ((uint)(nextByte - '0') > '9' - '0')
+            {
+                _lineBytePosition += i;
+                ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedDigitNotFound, nextByte);
+            }
+            i++;
+
+            return ConsumeIntegerDigits(ref data, ref i);
+        }
+
+        private ConsumeNumberResult ConsumeSign(ref ReadOnlySpan<byte> data, ref int i, byte nextByte)
+        {
+            if (i >= data.Length)
+            {
+                if (IsLastSpan)
+                {
+                    _lineBytePosition += i;
+                    ThrowHelper.ThrowJsonReaderException(ref this, ExceptionResource.ExpectedDigitNotFoundEndOfData, nextByte);
+                }
+                else
+                    return ConsumeNumberResult.NeedMoreData;
```

Comment:
Braces... I'll stop commenting on this, but we should use braces for all blocks of an if/else if/else if any of them has braces.

Reply:
OK, that is a fair distinction and compromise. Will make those changes since those are less relevant to the reasons I mentioned above: https://github.com/dotnet/corefx/pull/33216#discussion_r230530018